Common/Extensions.cs

```diff
@@ -230,6 +230,15 @@ public static long ToInt64(this string str)
         }
 
         /// <summary>
+        /// Check if a number is NaN or equal to zero
+        /// </summary>
+        /// <param name="value">The double value to check</param>
+        public static bool IsNaNOrZero(this double value)
+        {
+            return double.IsNaN(value) || Math.Abs(value) < double.Epsilon;
```

Comment:
#loveresharper

In other news, I try not to pollute our codebase with resharper hints, but there just for your information (and anyone following), there is a // Resharper Disable for floating point comparison with zero to remove the R# warning.